src/arrayMethodJoin.js

```diff
@@ -4,9 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    } else if (this.length === 1) {
+      return this[0] + '';
+    }
+
+    let string = '';
+    const usefulSeparator = '' + separator;
```

Comment:
If you add element to a string, it becomes string. No need to create usefulSeparator variable here